app/controllers/questions_controller.rb

```diff
@@ -33,7 +34,10 @@ def destroy
 private
 
   def find_quiz
-    @quiz = Quiz.find(params[:quiz_id])
+    @quiz = current_user.quizzes.find(params[:quiz_id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = "You are not authorised to modify this quiz."
+    redirect_to quiz_path(params[:quiz_id])
```

Comment:
This has a couple of issues: 

1) It doesn't check the user, so there is not an authorisation layer
2) It will flag any non-existent quiz as not authorised which would be incorrect.

If you think it's necessary to have a different response for authorisation to resource not found you should go back to a before_action. But it isn't necessary, for example if you view a private repository on GitHub in a private browsing session that returns a 404.

Reply:
Sure, you're probably a bit overly concerned with an edge case if you're down to personal preferences here. Doing a redirect isn't the correct semantic approach to either situation really as the page you're redirected to isn't the new URI of the content. You can learn more from the HTTP status codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status

Semantically, if you're not authorised to view something the status code should be 403 to indicate that the resource is forbidden from you. Though 404 is commonly used as it then leaves ambiguity as to whether the resource even exists - which can be good to not inform an actor